quai/p2p_backend.go

```diff
@@ -89,6 +89,8 @@ func (qbe *QuaiBackend) OnNewBroadcast(sourcePeer p2p.PeerID, topic string, data
 		// TODO: Determine if the block information was lively or stale and rate
 		// the peer accordingly
 		backend.WriteBlock(&data)
+		// If it was a good broadcast, mark the peer as lively
+		qbe.p2pBackend.MarkLivelyPeer(sourcePeer, topic)
```

Comment:
You should be checking if this broadcast was recent/"lively" data. If its a stale/latent broadcast, you wouldn't want to give this peer credit for being lively.... ofc its up to core to determine what qualifies as a "lively" response vs latent

Reply:
Yeah i agree, we will do that next, this is just moving the LivelyMarking to the correct spot, we were marking peers lively because they were broadcasting (could be even transactions).

Reply:
sounds good. Can you please create an issue?

Reply:
Its already an issue https://github.com/dominant-strategies/go-quai/issues/1763